Disable -pthread for compiler/linker on OSX

oderwat opened the pull request:
The -pthread is not needed on Darwin/OS X and the Apple compilers give a warning about this if you use --threads:on with the Nim compiler.

kirbyfan64 commented:
[I think you need it for compiling but not linking](http://stackoverflow.com/a/19382663/2097780).

Araq commented:
kirbyfan64 is correct. Also Apple compilers should shut up. Warnings about irrelevant crap suck.

oderwat commented:
It works perfectly without that it seems. I found that http://www.libusb.org/ticket/96 and your link too (which is about xcode 5, afaik it was not needed there either. I removed that often from stuff I wanted to compile). 

Funny enough it does not complain about passing it to the compiler `{.passC: "-pthread".}` but you get `clang: warning: argument unused during compilation: '-pthread'` if you pass it to the linker `{.passL: "-pthread".}`. I had a version where I just omit it for the linker but that made no difference in the exe (checksums).

Using gcc-4.9 does not work. but that is because if complains with an error about: `gcc: error: unrecognized command line option '-fasm-blocks'`.

oderwat commented:
Hint: @Araq complains about Apple complaining about irrelevant crap

Araq commented:
Why are warnings not disabled for clang anyway? The default config has: clang.options.always = "-w"

oderwat commented:
I have

```
@if macosx or freebsd:
  cc = clang
  tlsEmulation:on
  gcc.options.always = "-w -fasm-blocks"
  gcc.cpp.options.always = "-w -fasm-blocks -fpermissive"
@else:
  gcc.options.always = "-w"
  gcc.cpp.options.always = "-w -fpermissive"
@end
```
in my /etc/nim.cfg ... fine hand copied there because I do not install with sudo on my mac :) but it is there and the compiler hints that he loads it.

```
nim c --threads=on --parallelBuild:1 -r threadpool_1.nim
/private/etc/nim.cfg(47, 2) Hint: added path: '/Users/NotTheAraq/.babel/pkgs/' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/fuse-0.1.0' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/docopt-0.1.0' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/x11-1.0' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/argument_parser-0.4.0' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/badger_bits-0.2.4' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/genieos-9.4.3' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/compiler-0.10.3' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/nimble-0.6.0' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/nake-1.4' [Path]
/private/etc/nim.cfg(48, 2) Hint: added path: '/Users/NotTheAraq/.nimble/pkgs/' [Path]
Hint: used config file '/etc/nim.cfg' [Conf]
Hint: used config file '/Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/threadpool_1.nim.cfg' [Conf]
Hint: system [Processing]
Hint: threadpool_1 [Processing]
Hint: threadpool [Processing]
Hint: cpuinfo [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: os [Processing]
Hint: times [Processing]
Hint: posix [Processing]
Hint: cpuload [Processing]
Hint: locks [Processing]
lib/nim/core/locks.nim(22, 43) Warning: LockEffect is deprecated [Deprecated]
lib/nim/core/locks.nim(25, 44) Warning: LockEffect is deprecated [Deprecated]
threadpool_1.nim(17, 12) Hint: 'n' is declared but not used [XDeclaredButNotUsed]
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/threadpool_1.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/threadpool_1.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/system.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/system.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/threadpool.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/threadpool.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/cpuinfo.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/cpuinfo.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/strutils.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/strutils.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/parseutils.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/parseutils.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/os.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/os.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/times.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/times.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/posix.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/posix.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/cpuload.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/cpuload.c
clang -c  -w -pthread  -I/usr/local/lib/nim -o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/locks.o /Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/nimcache/locks.c
[Linking]
clang: warning: argument unused during compilation: '-pthread'
Hint: operation successful (18053 lines compiled; 0.542 sec total; 20.204MB) [SuccessX]
/Users/NotTheAraq/devsys/workspace/nim/nim-simples/src/parallel/threadpool_1
9227465
9227465
9227465
9227465
```

oderwat commented:
> Geee, guess what. I added that option just for fun back in the days. Apple's stupid GCC used to require it, now of course it doesn't even eat it anymore...

It is the "normal" GCC 4.9 (homebrew) which complains about that. Apples "strange" current system GCC and former GCC (4.2 which can be installed with homebrew to compile some old stuff) still eat it.

I am a bit busy for another project right now but I will try to get gcc 4.9 running. I also would love to get tcc running (in script mode) to use Nim as scripting language.
